### ISSUE TITLE
React Native Deep Link not working from email redirection

### ISSUE THREAD

**Issue by Muskan-Redaptive** [facebook/react-native#34318]
### Description

When I'm trying to redirect from email to the app using deep link host and scheme, it's redirecting me to the browser with that link and If I use same link in html file with href it's redirecting me to the app.

Anyone know about it?
Thanks in Advance!

### Version

0.68.1

### Output of `npx react-native info`

System:
    OS: macOS 12.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 101.83 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - /usr/local/bin/node
    Yarn: Not Found
    npm: 8.5.5 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 23, 28, 29, 30, 31, 32
      Build Tools: 29.0.2, 30.0.2, 31.0.0, 32.0.0, 32.1.0
      System Images: android-31 | Google APIs ARM 64 v8a, android-32 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8609683
    Xcode: 13.4/13F17a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.15 - /opt/homebrew/opt/openjdk@11/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: ^0.68.1 => 0.68.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

We are adding email template from backend in html file and there we are creating a activation link in href.
And using same host and scheme in intent filter in manifest file.

### Snack, code example, screenshot, or link to a repository

 <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <data android:scheme="https" android:host="my-installer-app"  android:pathPrefix="/example" />
 </intent-filter>

Using above code in manifest for deep linking 

And at backend using this url for redirecting through mail in app
https://my-installer-app/example

but it's redirecting me to the browser.

**Comment by H4mxa** [comment]
Any update on this? I'm facing the issue in android 12 and ios. After google authentication instead of redirecting me to the app it redirects me to browser.

**Comment by Muskan-Redaptive** [comment]
Yes, it was backend issue. App redirection link was not correct in email template.
